pkg/kubelet/pod_workers.go

```diff
@@ -31,14 +31,58 @@ import (
 	"k8s.io/kubernetes/pkg/util/wait"
 )
 
+// OnCompleteFunc is a function that is invoked when an operation completes.
+// If err is non-nil, the operation did not complete successfully.
+type OnCompleteFunc func(err error)
+
+// PodStatusFunc is a function that is invoked to generate a pod status.
+type PodStatusFunc func(pod *api.Pod, podStatus *kubecontainer.PodStatus) api.PodStatus
+
+// KillPodOptions are options when performing a pod update whose update type is kill.
+type KillPodOptions struct {
+	// PodStatusFunc is the function to invoke to set pod status in response to a kill request.
+	PodStatusFunc PodStatusFunc
+	// PodTerminationGracePeriodSecondsOverride is optional override to use if a pod is being killed as part of kill operation.
+	PodTerminationGracePeriodSecondsOverride *int64
+}
+
+// UpdatePodOptions is an options struct to pass to a UpdatePod operation.
+type UpdatePodOptions struct {
+	// pod to update
+	Pod *api.Pod
+	// the mirror pod for the pod to update, if it is a static pod
+	MirrorPod *api.Pod
+	// the type of update (create, update, sync, kill)
+	UpdateType kubetypes.SyncPodType
+	// optional callback function when operation completes
+	OnCompleteFunc OnCompleteFunc
```

Comment:
If we are keeping the general `OnCompletFunc` name, could you at least add some comments warning people that pod worker doesn't necessary process each and every update? It might skip to the latest update.